### ISSUE TITLE
Safe: Custom node Name should be deprecated or a new solution.

### ISSUE THREAD

**Issue by jiangfuyao** [paritytech/substrate-telemetry#147]
I think future the telemetry's custom node Name should be deprecated. Because, someone can set name: "polkadot-official"/ "authority"/ "coinbase" / "bitcoin.com" / "Binance"/ ... , Pretend to be the official node, there will be unfair ways to attract nominators. If it's large, it's a security risk.

**Comment by chevdor** [comment]
Hi @jiangfuyao, the names you see do not come from telemetry but from the polkadot/substrate nodes themselves.

Source: an old PR I made: you cannot name a polkadot/substrate node anything matching a url regexp so your .com example would not be allowed. I tested to be sure it is still the case:
```
$ polkadot --name bitcoin.com
Error: Invalid node name 'bitcoin.com'. Reason: Node name should not contain invalid chars such as '.' and '@'. If unsure, use none.
```

I kinda remember that there are sanity rules in telemetry itself so in the (actually no longer possible) event that a node shows up with a very long name, telemetry would truncate anyway for instance.

There are a few other rules to prevent name spamming. Sure you could name your validator `bitcoin_com` or `polkadot_official` but I doubt anyone would care. There is however indeed no mechanism to prevent using the other names you mention. There is also no mechanism to prevent duplicates. In other words, all the nodes could `--name` themselves polkadot-official if they wanted to.

Why do you think people would use telemetry as primary source for picking a validator? 
That sounds like the very wrong tool for the job.

Telemetry shows the validator´s address so people would have least need to double check and if they can double check, that means they have already the validator´s address and don´t really need telemetry in the first place.

Do you have an idea to solve the issue you raise?

I could imaging using  some kinda of onchain DNS and only allow names owned by the session/controller/stash. I am sure such DNS will become available but I am not sure the mechanics around protecting the name ppl pick for a node is worth the troubles.

I don´t want to give bad ideas so I will not list all the tricks you could do but in the end, whatever you do, you only let a a non clickable string show up in telemetry.

**Comment by jiangfuyao** [comment]
In a discussion meeting, one of the founders of PoS project said a troublesome thing: An official name for validator would attract numerous nominations. So, I open this issue.  
When people are linked to profits, they do whatever it takes to investigate. When they see telemetry has a name and an address, they will no doubt prefer some well-known ones.  
I think automatic shortening of addresses would be a better solution.  
Such as: `5Esmbgn75sFc9Q2upPtQHgmLtUJPShtEHBwHdmsPHforwLLN` => `FNV1`

**Comment by chevdor** [comment]
I agree, showing the address always would be good but bloats the UI so using short names could be a good solution.
